### PR TITLE
feat: update prometheus-operator dependency in go.mod

### PR DESCRIPTION
**Pull request by tremes**
This updates the prometheus-operator go dependency to v0.57.0

### CODE REVIEW

File: pkg/apis/monitoring/v1alpha1/types.go [hunk 1]
```diff
@@ -60,9 +60,8 @@ type MonitoringStackSpec struct {
 
 	// Time duration to retain data for. Default is '120h',
 	// and must match the regular expression `[0-9]+(ms|s|m|h|d|w|y)` (milliseconds seconds minutes hours days weeks years).
-	// +kubebuilder:validation:Pattern="^[0-9]+(ms|s|m|h|d|w|y)$"
```
Comment (tremes) [inline pkg/apis/monitoring/v1alpha1/types.go:63]: I hope this is safe to remove when there's validation at https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1/types.go#L385

Comment (jan--f) [reply pkg/apis/monitoring/v1alpha1/types.go:63]: Yeah that looks great, nice improvement!

**Review by jan--f — CHANGES_REQUESTED** [review]
Two places still need the update:

- [ ] The dependency resources need the bump: https://github.com/rhobs/observability-operator/blob/f83e38b5df749390a4212525ab601486c7e4c2da/deploy/dependencies/kustomization.yaml#L2-L5
- [ ] The mixins from prometheus-operator need the bump: https://github.com/rhobs/observability-operator/blob/f83e38b5df749390a4212525ab601486c7e4c2da/jsonnet/jsonnetfile.json#L11

**Review by jan--f — APPROVED** [review]
Thanks!

**Pull request merged**